llm/openai/openai.go

```diff
@@ -4,12 +4,15 @@ import (
 	"bytes"
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io"
 	"net/http"
 	"net/url"
 	"strings"
 
+	tools "github.com/primalmotion/simplai/llm/internal"
```

Comment:
tools is a confusing name. also don't put stuff directly in interrnal.

you should move this to internal/utils/utils.go